lib/src/components/NxDropdown/__tests__/NxDropdown.test.tsx

```diff
@@ -5,357 +5,301 @@
   });
 
-  it('does not call onToggleCollapse if a click happens anywhere when the dropdown is disabled', function() {
-    const onToggleCollapse = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse, isOpen: true, disabled: true }, { attachTo: container });
+  it('does not call onToggleCollapse if a click happens anywhere when the dropdown is disabled', async function() {
+    const user = userEvent.setup();
+    const onToggleCollapse = jest.fn();
+    quickRender({ onToggleCollapse, isOpen: true, disabled: true });
 
     expect(onToggleCollapse).not.toHaveBeenCalled();
-
-    act(() => {
-      document.dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    await user.click(document.body);
     expect(onToggleCollapse).not.toHaveBeenCalled();
   });
 
-  it('calls onToggleCollapse once when clicking to open the dropdown', function() {
-    const onToggleCollapse = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse }, { attachTo: container });
+  it('calls onToggleCollapse once when clicking to open the dropdown', async function() {
+    const user = userEvent.setup();
+    const onToggleCollapse = jest.fn();
 
-    expect(onToggleCollapse).not.toHaveBeenCalled();
+    quickRender({ onToggleCollapse });
 
-    act(() => {
-      component!.find(NxButton).getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    expect(onToggleCollapse).not.toHaveBeenCalled();
+    await user.click(screen.getByRole('button'));
     expect(onToggleCollapse).toHaveBeenCalledTimes(1);
   });
 
-  it('calls onToggleCollapse once when clicking to close the dropdown', function() {
-    const onToggleCollapse = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse, isOpen: true }, { attachTo: container });
+  it('calls onToggleCollapse once when clicking to close the dropdown', async function() {
+    const user = userEvent.setup();
+    const onToggleCollapse = jest.fn();
+    quickRender({ onToggleCollapse, isOpen: true });
 
     expect(onToggleCollapse).not.toHaveBeenCalled();
-
-    act(() => {
-      component!.find(NxButton).getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    component!.update();
+    await user.click(screen.getByRole('button'));
     expect(onToggleCollapse).toHaveBeenCalledTimes(1);
   });
 
   it('calls onToggleCollapse if ESC is pressed within the component while the dropdown is open', function() {
-    const onToggleCollapse = jest.fn(),
-        component = getMountedComponent({ onToggleCollapse, isOpen: true });
-
-    component.simulate('keyDown', { key: 'Escape', preventDefault: jest.fn() });
+    const onToggleCollapse = jest.fn();
+    const component = renderEl({ onToggleCollapse, isOpen: true })!;
+    fireEvent.keyDown(component, { key: 'Escape' });
```

Comment:
Is there a reason `userEvent` isn't being used here?

Reply:
> Is there a reason `userEvent` isn't being used here?

Oh because the event need to be fired within the component.
Should I focus on the toggle and use userEvent on that?

Reply:
> Should I focus on the toggle and use userEvent on that?

I think that'd be preferable yeah

Reply:
> > Should I focus on the toggle and use userEvent on that?
> 
> I think that'd be preferable yeah

Updated here: c79a3f815f7ed7b13d789a32448d3b5332fe1974